### PR TITLE
Added tutorial outline

### PR DESCRIPTION
**Pull request by Mierdin**

### CODE REVIEW

File: tutorials.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+StackStorm Tutorials
+--------------------
+
+In order to create a "StackStorm Path to Mastery" program, it's necessary to create some tutorials that users can follow in order to become proficient at each level
+
+Each tutorial should be composed primarily of text, and should read much like a blog post. Like any high-quality blog post, helpful illustrations and examples should be interspersed throughout, including code/shell snippets, images/diagrams, and video (screencasts, headshot explanations).
+
```
Comment (dzimine) [inline tutorials.md]: do  we carry one story throughout the whole series? Like "automate XYZ"? (and what is a good XYZ should be).

**Comment by samirsss** [comment]
Some quick feedback:
Is there some tutorial/guide to explain the Stackstorm terminology about packs, rules, sensors, actions before we ask users to code actions?

**Comment by dzimine** [comment]
Great, thanks! 

The break-down looks right.
The complexity: we can try "sizable pieces" which is for example 5 min read 15 min to follow (pick your numbers). In this way, 

- install is "one" (yes with st2vagrant getting st2 and upgrading to bwc is easy, the read adds references to advanced ways of installing for awareness but we continue to assume vagrant and OS for the rest of tutorial, if someone's special it's on him).
    - [ ] TODO: dzimine to find the video of installing and upgrading st2 with vagrant.

- actions - this needs to be split to more chunks one can swallow. First action, second action, python action. "3 units" easily. Same applies to others: break it up if needed to smaller units, and measure in units. 

- packs - if we are telling a story, we can spread it out. At the completion of the action, we can turn it into a pack, init, commit, and push to github. And continue using `pack install` to update it from dev or github upstream. (This will be  new content). 
    - DZ: cause IMO if you get to the workflows and your stuff is not yet in github, something is wrong. Even the initial setup: how & where do I update the action when I didn't get it right? Do I reload the whole shoebank when I change a line in action? (NO). In meta-data? (NO, `action update`) - but many people are asking these basic questions. 
       I discussed it with Manas 1 year ago and we voted it wasn't nessessary as folks like netflix had no problems figuring out their dev environment. Now that we cast a wider net, let's give one opinionated dev setup/workflow (as a separate chapter?) which is not the only one but assumed for the purpose of tutorial. Vagrant + shared folder + git operated on the host manchine, and pack install working with either `file:///` or github upstream.

**Comment by Mierdin** [comment]
From @emedvedev: "That's my main piece of feedback. 
The other one (directly from #learning) is that the tutorial "flow" isn't very consistent, I think. If we're talking about using StackStorm, and we start with "run some example actions", how'd we get those actions to begin with? Actions are a first step for _developing_ for StackStorm, but for _using_ it starting with packs is more logical. How would you even discover if StackStorm is a good fit for your scenario or you'll end up doing a lot of manual work? By looking at packs.
Also, I think creation could be separated from usage a bit more clearly. You're learning to use actions, sensors, rules, packs (not create them yet, judging from the description!), and then you're suddenly creating chains and workflows. That's a pretty big jump."

**Comment by Mierdin** [comment]
From @arma: 

I think before Install/Configure there could be a short introductory section to describe what StackStorm is at a high level, principles, infra as code, auto-remediation, DevOps and so on. Eg. what is st2, why it's useful and "what you'll learn"

**Comment by Mierdin** [comment]
@samirsss Added basic concepts section in https://github.com/Mierdin/st2-notes/pull/1/commits/ef90dca97ce55cb9163413b894db3e69926119ff based on your comment

**Comment by dzimine** [comment]
Building on top of @emedvedev's idea:
> You can use StackStorm even without being able to create a Python action, for example.
Yes
May be wish we had a good self-contained sample that was not boring like example, can live on #Exchange and user can download and hit the ground playing with, modifying. Too bad by the structure 

I am also +1 on down-playing or completely removing ActionChain at least for starter. We can at first use Mistral as action chain, and have "Mistral again" later to give more complex features.

**Comment by dzimine** [comment]
> So i'm not clear on whether these tutorials are more for users or developers 
For "coders". For automators. 
Developer assumes more than one needs to deal with stackstorm. 
Users? Coding is required to do anything with stackstorm. Our users are typical devops who know to use Linux, code python, search StackOverflow and have above-average curiosity.
There is no clear cut , but it's also a reality: pick Lindsay - is he a user? Pick someone who don't have Lindsay's skill - they don't qualify using stackstorm.

**Comment by Mierdin** [comment]
@dzimine I was more referring to "developers" as someone that would come in and write their own pack, or otherwise extend stackstorm. To @emedvedev point, there's a big difference between that and someone that just writes workflows for existing actions/packs. In this context, I'm referring to those as "users". So my main question is if these tutorials should include what I'm calling "developers", meaning they go deeper than most users of Stackstorm by writing their own extensions.

**Comment by dzimine** [comment]
The simple answer is "general users". 
But it's more nuanced. 

> To @emedvedev point, there's a big difference between that and someone that just writes workflows for existing actions/packs.

There is no way to write anything in st2 with just workflows + packs, without writing an action. Therefore, I am with @emedvedev when he says that we need to intro the action early. It may not be python action: REST or core_remote_script action may be just fine. 

May be, have a separate episode to write an integration pack to something.

Or, once you get your over-arching story, you'll see that there gotta be an integration with something.

**Comment by dzimine** [comment]
Idea: 
* Story 1: Reproduce st2-demo - log file clean-up. Actions there are minimal, can be done with just scripts. https://github.com/StackStorm/st2incubator/tree/master/packs/st2-demos 
* Story 2: how about "clean-up stall AWS instances"?

**Pull request merged**